langchain/document_loaders/evernote.py

```diff
@@ -5,78 +5,110 @@
-
-    note_dict["resource"] = resources
-
-    return note_dict
-
-
-def _parse_note_xml(xml_file: str) -> str:
-    """Parse Evernote xml."""
-    # Without huge_tree set to True, parser may complain about huge text node
-    # Try to recover, because there may be "&nbsp;", which will cause
-    # "XMLSyntaxError: Entity 'nbsp' not defined"
-    from lxml import etree
-
-    context = etree.iterparse(
-        xml_file, encoding="utf-8", strip_cdata=False, huge_tree=True, recover=True
-    )
-    result_string = ""
-    for action, elem in context:
-        if elem.tag == "note":
-            result_string += _parse_note(elem)["content"]
-    return result_string
-
-
 class EverNoteLoader(BaseLoader):
-    """Loader to load in EverNote files.."""
+    """EverNote Loader.
+    Loads an EverNote notebook export file e.g. my_notebook.enex into Documents. Instructions on producing this
+    file can be found at https://help.evernote.com/hc/en-us/articles/209005557-Export-notes-and-notebooks-as-ENEX-or-HTML
+
+    Args:
+        file_path (str): The path to the notebook export with a .enex extension
+    """
 
     def __init__(self, file_path: str):
         """Initialize with file path."""
         self.file_path = file_path
 
     def load(self) -> List[Document]:
-        """Load document from EverNote file."""
-        text = _parse_note_xml(self.file_path)
-        metadata = {"source": self.file_path}
-        return [Document(page_content=text, metadata=metadata)]
+        """Load documents from EverNote export file."""
+        return [
+            Document(
+                page_content=note["content"],
+                metadata={
+                    **{key: value for key, value in note.items() if key != "content"},
+                    **{"source": self.file_path}
+                })
+            for note in self._parse_note_xml(self.file_path) if note.get("content") is not None
+        ]
+
+    @staticmethod
+    def _parse_content(content: str) -> str:
+        try:
+            import html2text
+            return html2text.html2text(content)
+        except ImportError as e:
+            logging.error("Could not import `html2text`. Although it is not a required package to use Langchain, using the EverNote loader requires `html2text`. Please install `html2text` via `pip install html2text` and try again.")
+            raise e
+
+    @staticmethod
+    def _parse_resource(resource: list) -> dict:
+        rsc_dict: Dict[str, Any] = {}
+        for elem in resource:
+            if elem.tag == "data":
+                # Some times elem.text is None
+                rsc_dict[elem.tag] = b64decode(elem.text) if elem.text else b""
+                rsc_dict["hash"] = hashlib.md5(rsc_dict[elem.tag]).hexdigest()
+            else:
+                rsc_dict[elem.tag] = elem.text
+
+        return rsc_dict
+
+    @classmethod
+    def _parse_note(cls, note: List, prefix: Optional[str] = None) -> dict:
```

Comment:
It's much more common for classmethods to return instances of the class. 

Maybe make it into a staticmethod or else detach from class as python is a land of verbs. :)

Reply:
Yeah you are spot on, not sure what I was thinking, I've moved these to static methods which they should have been in the first place.